src/client/api/tests/eq_tests.c

```diff
@@ -1043,6 +1043,145 @@ eq_test_8()
 	return rc;
 }
 
+static bool	stop_progress;
+static int	polled_events;
+pthread_mutex_t	eqh_mutex;
+
+static void *
+th_eq_poll(void *arg)
+{
+	struct daos_event	*eps[EQT_EV_COUNT] = { 0 };
+
+	while (1) {
+		int rc;
+
+		if (stop_progress)
+			pthread_exit(NULL);
+
+		rc = daos_eq_poll(my_eqh, 0, DAOS_EQ_NOWAIT, EQT_EV_COUNT, eps);
+		if (rc < 0) {
+			print_error("EQ poll failed: %d\n", rc);
+			rc = -1;
+			pthread_exit(NULL);
+		}
+
+		if (rc) {
+			D_MUTEX_LOCK(&eqh_mutex);
+			polled_events += rc;
+			D_MUTEX_UNLOCK(&eqh_mutex);
+		}
+	}
+}
+
+static int
+eq_test_9()
+{
+	struct daos_event	*events[EQT_EV_COUNT] = { 0 };
+	struct daos_event	*eps[EQT_EV_COUNT] = { 0 };
```

Comment:
EQT_EV_COUNT is defined as 1000, maybe not enough to test possible race condition. For larger number, may need to allocate memory.

Reply:
yea sure, we can have a different PR for that. we've seen issues before on VM runs when the workload is high, so we can do a test to increase that there.